build-farm/set-platform-specific-configurations.sh

```diff
@@ -25,5 +25,60 @@ fi
 
 export VARIANT_ARG="--build-variant ${VARIANT}"
 
-# shellcheck disable=SC1091,SC1090
-source "$SCRIPT_DIR/platform-specific-configurations/${OPERATING_SYSTEM}.sh"
\ No newline at end of file
+# Create placeholders for curl --output
+if [ ! -d "$SCRIPT_DIR/platform-specific-configurations" ]
+then
+    mkdir "$SCRIPT_DIR/platform-specific-configurations"
+fi
+if [ ! -f "$SCRIPT_DIR/platform-specific-configurations/platformConfigFile.sh" ]
+then
+    touch "$SCRIPT_DIR/platform-specific-configurations/platformConfigFile.sh"
+fi
+PLATFORM_CONFIG_FILEPATH="$SCRIPT_DIR/platform-specific-configurations/platformConfigFile.sh"
+
+# Setup for platform config download
+rawGithubSource="https://raw.githubusercontent.com"
+ret=0
+fileContents=""
+
+# Uses curl to download the platform config file
+# param 1: LOCATION - Repo path to where the file is located
+# param 2: SUFFIX - Operating system to append
+function downloadPlatformConfigFile () {
+    echo "Attempting to download platform configuration file from ${rawGithubSource}/$1/$2"
+    # make-adopt-build-farm.sh has 'set -e'. We need to disable that for the fallback mechanism, as downloading might fail
+    set +e
+    curl "${rawGithubSource}/$1/$2" > "${PLATFORM_CONFIG_FILEPATH}"
+    ret=$?
+    # A download will succeed if location is a directory, so we also check the contents are valid
+    fileContents=$(cat $PLATFORM_CONFIG_FILEPATH)
+    set -e
+}
+
+# Attempt to download and source the user's custom platform config
+downloadPlatformConfigFile "${PLATFORM_CONFIG_LOCATION}" ""
+# Regex to spot github api error messages similar to "404: Not Found"
+contentsErrorRegex="40[0-9]: [a-zA-Z ]+"
+
+if [ $ret -ne 0 ] || [[ $fileContents =~ $contentsErrorRegex ]]
+then
+    # Check to make sure that a OS file doesn't exist if we can't find a config file from the direct link
+    echo "[WARNING] Failed to find a user configuration file, ${rawGithubSource}/${PLATFORM_CONFIG_LOCATION} is likely a directory so we will try and search for a ${OPERATING_SYSTEM}.sh file."
```

Comment:
Could extract this block to a function

Reply:
The block is already using functions to do the important bits (i.e. downloading and setting shell to accept errors). I personally feel it would be overcomplicated to further functionize this file?

Reply:
Yeah that's fine we can look at it again later.